src/js/models/panel_model.js

```diff
@@ -301,25 +313,56 @@
                 m = pad(parseInt((deltaT % 3600) / 60));
                 s = pad(Math.round(deltaT % 60));
                 text = h + ":" + m + ":" + s;
+            } else { // Format unknown
+                return ""
             }
             if (["0 s", "0:00", "0 mins", "0:00:00"].indexOf(text) > -1) {
                 isNegative = false;
             }
             return (isNegative ? '-' : '') + text;
         },
 
-        create_labels_from_time: function(options) {
-            
-            this.add_labels([{
-                    'time': options.format,
-                    'size': options.size,
-                    'position': options.position,
-                    'color': options.color
-            }]);
+		get_name_label_text: function(format) {
+            var text = "";
+            if (format === "image") {
+                var pathnames = this.get('name').split('/');
+                text = pathnames[pathnames.length-1];
+            } else if (format === "dataset"){
+                text = this.get('datasetName') ? this.get('datasetName') : "No/Many Datasets";
+            }
+            return text;
+        },
+		
+		get_roi_label_text: function(format) {
```

Comment:
The indentation looks wrong here - I think you have Tabs instead of Spaces. Same in a bunch of other places. It looks OK in my editor but still best to keep it consistent - thx